app/workers/build_index_page_worker.rb

```diff
@@ -21,6 +21,8 @@ def perform(index_namespace, index_name, page_number, page_size)
 
   def serialize_for(operation, index, entries)
     entries.each_with_object([]) do |model, memo|
+      next if index.skip?(model)
+
```

Comment:
I was thinking we could use a "Tell Don’t Ask" approach.

instead of asking what we should skip,
we could tell what we want:

```ruby
get_ungrupped_headings(entries) do |model, memo| 
  # same stuff as before
end

def get_ungrupped_headings # I am sure you can find a better name :) 
   # filter out the grupped ones
end
```

Reply:
So I'd need the dataset to do a union on headings that we want plus everything else. In this case, asking is actually easier